### PR TITLE
Fix for 1258, return properties for top level packages

### PR DESCRIPTION
**Pull request by abpiskunov**
**Customer scenario**

Properties for top level packages were missing. They were broken by addin DT build for PackageReference xaml rule. Logic in References and Dependencies nodes relied on fact that all unresolved references were not coming from DT build and all resolved did come from DT build. 

Change is to bring all dependency rules to consistent persistence. Now all resolved xaml rules should have persistence ResolvedReference. This would help to differentiate between rules having same ItemType.

**Bugs this fixes:**

https://github.com/dotnet/roslyn-project-system/issues/1258

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

**Risk**

minimal

### CODE REVIEW

**Comment by abpiskunov** [comment]
@dotnet/project-system @srivatsn @333fred @davkean

**Comment by abpiskunov** [comment]
This change also fixes https://github.com/dotnet/roslyn-project-system/issues/1153

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ResolvedAnalyzerReference.xaml [hunk 1]
```diff
@@ -7,7 +7,7 @@
     Description="Resolved Analyzer reference properties"
     xmlns="http://schemas.microsoft.com/build/2009/properties">
     <Rule.DataSource>
-        <DataSource Persistence="ProjectFile" ItemType="Analyzer" HasConfigurationCondition="False" SourceOfDefaultValue="AfterContext" />
+        <DataSource Persistence="ResolvedReference" ItemType="Analyzer" HasConfigurationCondition="False" SourceOfDefaultValue="AfterContext" />
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/ResolvedAnalyzerReference.xaml]: Need to run https://github.com/nguerrera/xliff-converter to update all the localized rules as well.

**Comment by srivatsn** [comment]
Tagging @MattGertz for RTM

**Comment by natidea** [comment]
Approved to merge via [377253](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/377253)

**Pull request merged**